Python 3.9 tests are all failing with an obscure pkg_resources error

Example: https://github.com/pangeo-forge/pangeo-forge-recipes/runs/8164237694?check_suite_focus=true

```
E               pkg_resources.ContextualVersionConflict: (tornado 6.2 (/usr/share/miniconda3/envs/pangeo-forge-
recipes/lib/python3.9/site-packages), Requirement.parse('tornado<6.2,>=6.0.3'), {'distributed'})
```

--- a/.github/dependabot.yml
+++ b/.github/dependabot.yml
@@ -4,8 +4,8 @@ updates:
   - package-ecosystem: pip
     directory: "/"
     schedule:
-      interval: "daily"
+      interval: "weekly"
   - package-ecosystem: "github-actions"
     directory: "/"
     schedule:
-      interval: "daily"
+      interval: "weekly"

--- a/.github/workflows/main.yaml
+++ b/.github/workflows/main.yaml
@@ -140,7 +140,7 @@ jobs:
           python -m pip install --no-deps -e  .
       - name: 🏄‍♂️ Run Tests
         run: |
-          py.test tests -v -m ${{ matrix.pytest-mark }} \
+          pytest tests -v -m ${{ matrix.pytest-mark }} \
             --cov=pangeo_forge_recipes --cov-config .coveragerc \
             --cov-report term-missing \
             --cov-report xml \

--- a/.github/workflows/main.yaml
+++ b/.github/workflows/main.yaml
@@ -51,6 +51,9 @@ env:
 jobs:
   prepare-env:
     runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l {0}
     strategy:
       fail-fast: false
       matrix:
@@ -90,13 +93,15 @@ jobs:
         if: matrix.dependencies == 'upstream-dev'
         run: mamba env update -n pangeo-forge-recipes -f ci/upstream-dev.yml
       - name: 🐍 List conda env
-        shell: bash -l {0}
         run: |
           conda info
           conda list
   run-tests:
     needs: prepare-env
     runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l {0}
     strategy:
       fail-fast: false
       matrix:
@@ -131,11 +136,9 @@ jobs:
       - name: 🎯 Set path to include conda python
         run: echo "/usr/share/miniconda3/envs/pangeo-forge-recipes/bin" >> $GITHUB_PATH
       - name: 🌈 Install pangeo-forge-recipes package
-        shell: bash -l {0}
         run: |
           python -m pip install --no-deps -e  .
       - name: 🏄‍♂️ Run Tests
-        shell: bash -l {0}
         run: |
           py.test tests -v -m ${{ matrix.pytest-mark }} \
             --cov=pangeo_forge_recipes --cov-config .coveragerc \

--- a/.github/workflows/main.yaml
+++ b/.github/workflows/main.yaml
@@ -60,7 +60,7 @@ jobs:
         python-version: [3.8, 3.9]
         dependencies: ["releases-only", "upstream-dev"]
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - name: 🔁 Setup Python
         uses: actions/setup-python@v2
         with:
@@ -80,7 +80,7 @@ jobs:
       - name: 🎯 Set environment file
         id: env-file
         run: echo "env_file=ci/py${{ matrix.python-version }}.yml" >> $GITHUB_ENV
-      - uses: actions/cache@v2
+      - uses: actions/cache@v3
         name: 🗃 Cache environment
         with:
           path: /usr/share/miniconda3/envs/pangeo-forge-recipes
@@ -116,7 +116,7 @@ jobs:
           - dependencies: "upstream-dev"
             pytest-mark: "executor_generator"
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - name: 🎯 Set cache number
         id: cache-number
         # cache will last 3 days by default
@@ -124,7 +124,7 @@ jobs:
       - name: 🎯 Set environment file
         id: env-file
         run: echo "env_file=ci/py${{ matrix.python-version }}.yml" >> $GITHUB_ENV
-      - uses: actions/cache@v2
+      - uses: actions/cache@v3
         name: 🗃 Loaded Cached environment
         with:
           path: /usr/share/miniconda3/envs/pangeo-forge-recipes

--- a/.github/workflows/release.yaml
+++ b/.github/workflows/release.yaml
@@ -9,9 +9,9 @@ jobs:
   deploy:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - name: Set up Python
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v3
         with:
           python-version: '3.x'
       - name: Install dependencies

--- a/.github/workflows/tutorials.yaml
+++ b/.github/workflows/tutorials.yaml
@@ -19,6 +19,9 @@ jobs:
   test-tutuorials:
     name: 📓 Build Tutorial Notebook
     runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l {0}
     strategy:
       fail-fast: false
       matrix:
@@ -30,9 +33,9 @@ jobs:
          - nb-path: "xarray_zarr/opendap_subset_recipe.ipynb"
          - nb-path: "xarray_zarr/terraclimate.ipynb"
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
         if: ${{ github.event_name != 'repository_dispatch' }}
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
         if: ${{ github.event_name == 'repository_dispatch' }}
         with:
           fetch-depth: 0
@@ -57,7 +60,7 @@ jobs:
           details_url: https://github.com/${{github.repository}}/actions/runs/${{github.run_id}}
 
       - name: 🔁 Setup Python
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v3
         with:
           python-version: ${{ env.python-version }}
           architecture: x64
@@ -75,7 +78,7 @@ jobs:
       - name: 🎯 Set environment file
         id: env-file
         run: echo "env_file=ci/py${{ env.python-version }}.yml" >> $GITHUB_ENV
-      - uses: actions/cache@v2
+      - uses: actions/cache@v3
         name: 🗃 Cache environment
         with:
           path: /usr/share/miniconda3/envs/pangeo-forge-recipes
@@ -85,12 +88,10 @@ jobs:
         run: mamba env update -n pangeo-forge-recipes -f ${{ env.env_file }}
         if: steps.conda-cache.outputs.cache-hit != 'true'
       - name: 🐍 List conda env
-        shell: bash -l {0}
         run: |
           conda info
           conda list
       - name: 🌈 Install pangeo-forge-recipes package
-        shell: bash -l {0}
         run: |
           python -m pip install --no-deps -e  .
 
@@ -98,7 +99,6 @@ jobs:
       # Could we consolidate it somehow?
 
       - name: 🏗 Build Notebook
-        shell: bash -l {0}
         run: >
           pytest --timeout=3000 --nbmake docs/pangeo_forge_recipes/tutorials/${{ matrix.nb-path }}
 

--- a/ci/py3.9.yml
+++ b/ci/py3.9.yml
@@ -23,6 +23,7 @@ dependencies:
   - kerchunk>=0.0.6
   - lxml    # Optional dep of pydap
   - matplotlib  # needed for building tutorial notebooks
+  - nbmake>=1.3.0  # used in tutorial nb worklow
   - netcdf4
   - numcodecs
   - numpy
@@ -48,5 +49,5 @@ dependencies:
   - xarray>=0.18.0
   - zarr>=2.6.0
   - pip:
-    - nbmake>=1.3.0  # used in tutorial nb worklow
+
     - pytest-timeout

